### DISCUSSION TITLE
bash error while install fish using brew on Ubuntu

### DISCUSSION THREAD

**Discussion by Trid-collab** [Homebrew/discussions#4022] [Linux]
I have trying to install fish using fish using brew in Ubuntu 22.04
The installation finshes without error
but when I try to run fish it give me the following error

`-bash: /home/linuxbrew/.linuxbrew/bin/fish: Bad address`

Could you help directing me to the resolution. I am new to fish and installing for the first time. I guess I am missing something here

**Comment by SMillerDev** [comment]
What does `brew doctor` and `brew config` say?

**Reply by Trid-collab** [reply]
I am running it on latest WSL2 after updating with wsl --update on Ubuntu 22.04. The Win10 version on my laptop running the latest wsl2 is win10 version 21H2. Not sure it it has anything to do with it

brew config

`HOMEBREW_VERSION: 3.6.14
ORIGIN: https://github.com/Homebrew/brew
HEAD: 001bacee18c512657e1535dff757c8cfdaefc116
Last commit: 3 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 9585c1a9737c5d321383659bf83cffb9d83d7d5d
Core tap last commit: 2 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CURLRC: set
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: octa-core 64-bit unknown_0x6_0x8c
Clang: N/A
Git: 2.34.1 => /bin/git
Curl: 7.81.0 => /bin/curl
Kernel: Linux 4.4.0-19041-Microsoft x86_64 GNU/Linux
OS: Ubuntu 22.04.1 LTS (jammy)
Host glibc: 2.35
/usr/bin/gcc: 11.3.0
/usr/bin/ruby: N/A
glibc: N/A
gcc@11: N/A
gcc: 12.2.0
xorg: N/A`

 brew doctor
`Your system is ready to brew.`

**Reply by SMillerDev** [reply]
Did you install WSL from the windows store?

**Reply by Bo98** [reply]
> `Linux 4.4.0-19041-Microsoft x86_64 GNU/Linux`

You're still using WSL1 here.

**Reply by Trid-collab** [reply]
Thank you. Got it working. Had to deinstall the install WSL though it was wsl2 , then upgraded win 10 to 22H2, resinstalled WSL2 and updated it by running wsl --update
